src/posix/client.c

```diff
@@ -67,6 +67,98 @@ static void InputCallback(char *aLine)
 }
 #endif // OPENTHREAD_USE_READLINE
 
+static bool findDone(int *aDoneState, char aNowCharacter)
+{
+    bool found = false;
+
+    switch (aNowCharacter)
+    {
+    case 'D':
+        *aDoneState = *aDoneState == 0 ? 1 : -1;
+        break;
+    case 'o':
+        *aDoneState = *aDoneState == 1 ? 2 : -1;
+        break;
+    case 'n':
+        *aDoneState = *aDoneState == 2 ? 3 : -1;
+        break;
+    case 'e':
+        *aDoneState = *aDoneState == 3 ? 4 : -1;
+        break;
+    case '\r':
+    case '\n':
+        if (*aDoneState == 4)
+        {
+            found = true;
+        }
+        *aDoneState = 0;
+        break;
+    }
+    
+    return found;
+}
+
+static bool findError(int *aErrorState, char aNowCharacter)
+{
+    switch (aNowCharacter)
+    {
+    case 'E':
+        *aErrorState = *aErrorState == 0 ? 1 : -1;
+        break;
+    case 'r':
+        if (*aErrorState == 1 || *aErrorState == 2 || *aErrorState == 4)
+        {
+            (*aErrorState)++;
+        }
+        else
+        {
+            *aErrorState = -1;
+        }
+        break;
+    case 'o':
+        *aErrorState = *aErrorState == 3 ? 4 : -1;
+        break;
+    case '\r':
+    case '\n':
+        *aErrorState = 0;
+        break;
+    }
+
+    return *aErrorState == 5;
```

Comment:
```suggestion
    return *aErrorState == 6;
```
or use the same style as `FindDone`.